native/src/mc_interface/mc_interface_dump.h

```diff
@@ -0,0 +1,255 @@
+    W_UINT32(chunk->format);
+    W_FLOAT(chunk->x);
+    W_FLOAT(chunk->y);
+    W_FLOAT(chunk->z);
+    W_UINT32(chunk->id);
+    W_UINT32(chunk->vertex_buffer_size);
+    W_UINT32(chunk->index_buffer_size);
+    dump_out.write((char*) chunk->vertex_data, chunk->vertex_buffer_size);
+    dump_out.write((char*) chunk->indices, chunk->index_buffer_size * sizeof(uint32_t));
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_execute_frame() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+}
+
+void dump_set_fullscreen(int fs) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    W_UINT32(fs);
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_should_close() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+}
+
+void dump_display_is_active() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+}
+
+void dump_add_gui_geometry(const char * geo_type, mc_gui_geometry * gui_geometry) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_out.write((char*) geo_type, strlen(geo_type) + 1);
+    dump_out.write((char*) gui_geometry->texture_name, strlen(gui_geometry->texture_name) + 1);
+    dump_out.write((char*) gui_geometry->atlas_name, strlen(gui_geometry->atlas_name) + 1);
+    W_UINT32(gui_geometry->index_buffer_size);
+    W_UINT32(gui_geometry->vertex_buffer_size);
+    dump_out.write((char*) gui_geometry->index_buffer, gui_geometry->index_buffer_size * sizeof(uint32_t));
+    dump_out.write((char*) gui_geometry->vertex_buffer, gui_geometry->vertex_buffer_size * sizeof(float));
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_clear_gui_buffers() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+}
+
+void dump_set_string_setting(const char * setting_name, const char * setting_value) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_out.write((char*) setting_name, strlen(setting_name) + 1);
+    dump_out.write((char*) setting_value, strlen(setting_value) + 1);
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_set_float_setting(const char * setting_name, float setting_value) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_out.write((char*) setting_name, strlen(setting_name) + 1);
+    W_FLOAT(setting_value);
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_set_player_camera_transform(double x, double y, double z, float yaw, float pitch) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    W_DOUBLE(x);
+    W_DOUBLE(y);
+    W_DOUBLE(z);
+    W_FLOAT(yaw);
+    W_FLOAT(pitch);
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_set_mouse_grabbed(int grabbed) {
+    dump_lock.lock();
+    DUMP_FUNC;
+    W_UINT32(grabbed);
+    dump_out.flush();
+    dump_lock.unlock();
+}
+
+void dump_get_materials_and_filters() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+}
+
+void dump_destruct() {
+    dump_lock.lock();
+    DUMP_FUNC;
+    dump_lock.unlock();
+    dump_out.close();
+}
+
+#else
```

Comment:
I know the way you did it is less code, but with that big #ifded ... #else, every time you add something to the mc interface, you also need to make sure it's added to both branches of this ifdef. Consider only doing the #ifdef on the function body, not all the functions. More code, but it's not as easy to break it without noticing it.

Reply:
Fixed